fiftyone/core/odm/mixins.py

```diff
@@ -141,16 +142,26 @@ def set_field(
         chunks = field_name.split(".", 1)
 
         if len(chunks) > 1:
-            doc = self.get_field(chunks[0])
+            root = chunks[0]
+            doc = self.get_field(root)
+
+            if doc is None:
+                doc = DynamicEmbeddedDocument()
+                self[root] = doc
+                self._changed_fields.remove(root)
```

Comment:
should we respect the 1create` parameter here?

Reply:
good call, updated 👍